### ISSUE TITLE
[akka-http-circe] Any reason for ErrorAccumulatingUnmarshaller to be private?

### ISSUE THREAD

**Issue by prascuna** [hseeberger/akka-http-json#157]
Hi,
I'm trying to write a rejection handler that will customise the json response in case more than one field is wrong.
The problem is that ErrorAccumulatingUnmarshaller is private, therefore I cannot pattern match on it

```scala
private val rejectionHandler: RejectionHandler = RejectionHandler.newBuilder()
    .handle {
      case r@MalformedRequestContentRejection(message, cause) =>
        val errors: List[String] = cause match {
          // Single error - This happens when mixing in FailFastCirceSupport
          case DecodingFailure(msg, ops) =>
            ops.map {
              case DownField(key) => s"Invalid format for field: ${key}"
              case o => s"An error has occurred: $o"
            }
          // Multiple errors - This happens when mixing in ErrorAccumulatingCirceSupport
          // I would like to pattern match but I can't because ErrorAccumulatingUnmarshaller is private
          // case e:ErrorAccumulatingUnmarshaller.DecodingFailures =>

          case e =>
            List("an unexpected error has happened " + e.getMessage)
        }
        complete(BadRequest, errors.mkString("\n"))
    }.result()
```

Is there another way to achieve that? If not, is there a reason for it to be private? 

Thanks,
Paolo

**Comment by hseeberger** [comment]
Good catch. Probably we need more API here. Proposal and PR welcome.

**Comment by prascuna** [comment]
More than happy to raise a PR.
I was thinking about two options, but maybe you had something else in mind:
a) just make `ErrorAccumulatingCirceSupport` object public and possibly `DecodingFailures` a case class to get the extractor for free
b) move `DecodingFailures` inside `BaseCirceSupport` and make sure that both the standard unmarshaller and the error accumulating one throw a `DecodingFailures` in case of failures. Obviously the standard unmarshaller will put only one `DecodingFailure` inside the list. This way, we can handle the failures in a common way, regardless if the end user is using the error accumulating feature or not.

How does it sound?

**Comment by hseeberger** [comment]
I suggest a) with the following modification: Move `DecodingFailures` to `object ErrorAccumulatingCirceSupport` and delete the now obsolete `object ErrorAccumulatingUnmarshaller`.

**Comment by hseeberger** [comment]
Closed by #159.
